src/Microsoft.ComponentDetection.Detectors/pip/PythonResolver.cs

```diff
@@ -208,12 +208,12 @@ private void AddGraphNode(PythonResolverState state, PipGraphNode parent, string
 
         private class PythonResolverState
         {
-            public readonly IDictionary<string, SortedDictionary<string, IList<PythonProjectRelease>>> ValidVersionMap
+            public IDictionary<string, SortedDictionary<string, IList<PythonProjectRelease>>> ValidVersionMap { get; }
                 = new Dictionary<string, SortedDictionary<string, IList<PythonProjectRelease>>>(StringComparer.OrdinalIgnoreCase);
 
-            public readonly Queue<(string, PipDependencySpecification)> ProcessingQueue = new Queue<(string, PipDependencySpecification)>();
+            public Queue<(string, PipDependencySpecification)> ProcessingQueue { get; } = new Queue<(string, PipDependencySpecification)>();
 
-            public readonly IDictionary<string, PipGraphNode> NodeReferences = new Dictionary<string, PipGraphNode>(StringComparer.OrdinalIgnoreCase);
+            public IDictionary<string, PipGraphNode> NodeReferences { get; } = new Dictionary<string, PipGraphNode>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Why remove `readonly`?

Reply:
it was necessary to convert field into a property to resolve [SA1401 : field should be private](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1401.md). And read-only is not valid on a property.